inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/service/RecommendationServiceImpl.java

```diff
@@ -1470,7 +1470,7 @@ public void calculateVisibility(CAS aCas, String aUser, AnnotationLayer aLayer,
                     for (AnnotationFS annotation : annotations.get(oi.getB())) {
                         String label = annotation.getFeatureValueAsString(feat);
                         for (AnnotationSuggestion suggestion : group) {
-                            if (!aLayer.isAllowStacking()
+                            if ((!aLayer.isAllowStacking() && label != null)
```

Comment:
`label != null` appears again in a conjunction in the next line. This redundancy can be removed by joining the two conditions.

Reply:
Done